### PR TITLE
[include-mapping] Python fixes

### PR DESCRIPTION
**Pull request by porglezomp**
I had to regenerate the include mapping while resolving a downstream merge conflict, and ran into two issue on my machine. These shouldn't change anything, just make things work on my config.

- Move the `multiprocessing.Pool` initializer to a top-level function, it was previously causing a pickle failure with my machine's python, specifically complaining about having to pickle the lambda or a local function. Works fine with a top-level function.
- Change the `env python` to `env python3` for convenience. The interpreter is installed as python3 on macOS.

It looks like about 2/3 of the scripts in `clang/tools/` are using `env python3` over `env python` so the latter seems a reasonable change to make.

### CODE REVIEW

File: clang/tools/include-mapping/cppreference_parser.py
Comment (hokein) [inline clang/tools/include-mapping/cppreference_parser.py:1]: can you also update this file to use `python3`?

**Review by hokein — COMMENTED** [review]
thanks, the change looks reasonable.

File: clang/tools/include-mapping/cppreference_parser.py
Comment (porglezomp) [reply clang/tools/include-mapping/cppreference_parser.py:1]: Done

**Review by hokein — APPROVED** [review]
thanks.

**Pull request merged**